### ISSUE TITLE
run Graph-Unet on PPI dataset

### ISSUE THREAD

**Issue by fatemehteimury** [rusty1s/pytorch_geometric#1089]
## ❓ Questions & Help
Hi, 
I'm trying to use Graph-Unet for PPI dataset and I change the original code. Unfortunately, I'm getting validation and test accuracy 0. 

This is the code I'm using right now. Also, I used different values for learning rate, hidden units, .. and there was no change in the final result. I'm pretty sure there's a bug in my code. hope you help me to find it. 

Thanks

This is the code I'm using right now:

```python
import os.path as osp
import os
import torch
import torch.nn.functional as F
from torch_geometric.datasets import PPI
from torch_geometric.data import DataLoader
#from torch_geometric.nn import GATConv
from sklearn.metrics import f1_score
from torch_geometric.nn import GraphUNet
from torch_geometric.utils import dropout_adj
os.environ['CUDA_VISIBLE_DEVICES'] = '3'


path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'PPI')
train_dataset = PPI(path, split='train')
val_dataset = PPI(path, split='val')
test_dataset = PPI(path, split='test')
train_loader = DataLoader(train_dataset, batch_size=1, shuffle=True)
val_loader = DataLoader(val_dataset, batch_size=2, shuffle=False)
test_loader = DataLoader(test_dataset, batch_size=2, shuffle=False)



class Net(torch.nn.Module):
    def __init__(self):        
        super(Net, self).__init__()
        pool_ratios = [.75, .5]
        self.unet = GraphUNet(train_dataset.num_features, 64, train_dataset.num_classes,
                              depth=3, pool_ratios=pool_ratios)

    def forward(self ,x, edge_index):
        edge_index, _ = dropout_adj(edge_index, p=0.2,
                                    force_undirected=True,
                                    num_nodes=int(x.shape[0]),
                                    training=self.training)
        x = F.dropout(x, p=0.001, training=self.training)
        x = self.unet(x, edge_index)
        return F.log_softmax(x, dim=1)
        

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
model = Net().to(device)
loss_op = torch.nn.BCEWithLogitsLoss()
optimizer = torch.optim.Adam(model.parameters(), lr=0.0005)


def train():
    model.train()

    total_loss = 0
    for data in train_loader:
        num_graphs = data.num_graphs
        data.batch = None
        data = data.to(device)
        optimizer.zero_grad()
        loss = loss_op(model(data.x, data.edge_index), data.y)
        total_loss += loss.item() * num_graphs
        loss.backward()
        optimizer.step()
    return total_loss / len(train_loader.dataset)


def test(loader):
    model.eval()
#
    ys, preds = [], []
    for data in loader:
        ys.append(data.y)
        with torch.no_grad():
            out = model(data.x.to(device), data.edge_index.to(device))
        preds.append((out > 0).float().cpu())

    y, pred = torch.cat(ys, dim=0).numpy(), torch.cat(preds, dim=0).numpy()
    return f1_score(y, pred, average='micro') if pred.sum() > 0 else 0


for epoch in range(1, 101):
    loss = train()
    val_f1 = test(val_loader)
    test_f1 = test(test_loader)
    print('Epoch: {:02d}, Loss: {:.4f}, Val: {:.4f}, Test: {:.4f}'.format(
        epoch, loss, val_f1, test_f1))
```

**Comment by rusty1s** [comment]
You need to remove the `log_softmax` call.

**Comment by fatemehteimury** [comment]
it's working right now. Thanks a lot
